src/ET_Client.php

```diff
@@ -119,6 +118,20 @@ function __construct($getWSDL = false, $debug = false, $params = null)
 					$this->baseSoapUrl = $this->defaultBaseSoapUrl;
 				}
 			}
+			if(array_key_exists('useOAuth2Authentication', $config)){$this->useOAuth2Authentication = $config['useOAuth2Authentication'];}
+			if(array_key_exists("baseUrl", $config))
+			{
+				$this->baseUrl = $config["baseUrl"];
+			}
+			else
+			{
+				if($this->useOAuth2Authentication === false) {
+					throw new Exception("baseUrl is null: Must be provided in config file when instantiating ET_Client");
+				}
+			}
+            if (array_key_exists('accountId', $config)){$this->accountId = $config['accountId'];}
```

Comment:
is it repeated?

Reply:
This is the PHP way of checking a key exists in an array. I don't think we can simplify this even more without a major refactoring.